src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -1889,6 +1928,40 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 
 	//#endregion
 
+	//#region Editor Actions
+
+	getEditorActions(): { actions: IToolbarActions; onDidChange: Event<IMenuChangeEvent> } {
+		const primary: IAction[] = [];
+		const secondary: IAction[] = [];
+
+		// Dispose previous listeners
+		this.editorToolBarMenuDisposables.clear();
```

Comment:
I think we need to change this so that the one who calls the method owns the lifecycle. Otherwise, now if you have 2 components that want `getEditorActions`, the one would dispose the actions from the other. So I think we need to also return a `IDisposable` from this method and use it in places where we call this method.